tests/tt_eager/python_api_testing/unit_testing/misc/test_scaled_dot_product_attention_decode.py

```diff
@@ -0,0 +1,660 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import torch
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import (
+    comp_allclose,
+    comp_pcc,
+)
+import tt_lib
+import ttnn
+from loguru import logger
+import pytest
+from models.utility_functions import skip_for_grayskull, skip_for_wormhole_b0
+import math
+
+
+def is_watcher_enabled():
+    return os.environ.get("TT_METAL_WATCHER") is not None
+
+
+def nearest_n(x, n):
+    return ((x + n - 1) // n) * n
+
+
+def nearest_pow_2(x):
+    if x < 1:
+        raise ValueError("x must be >= 1")
+    import math
+
+    power = math.ceil(math.log2(x))
+    return 1 << power
+    # if (2**math.log2(x) == x):
+    #     return x
+    # return 2**(int(x).bit_length())
+
+
+def num_to_corerange(x):
+    assert x < 8 or x % 8 == 0
+    num_x = min(x, 8)
+    num_y = x // num_x
+    assert num_x * num_y == x
+    return ttnn.experimental.tensor.CoreRange(
+        ttnn.experimental.tensor.CoreCoord(0, 0),
+        ttnn.experimental.tensor.CoreCoord(num_x - 1, num_y - 1),
+    )
+
+
+def get_chunk_size(s):
+    # Got to test this!
```

Comment:
If this is the best set of chunk sizes by sequence length, removing comment would be fine

Reply:
changed